src/agentscope/service/web/tripadvisor.py

```diff
@@ -0,0 +1,256 @@
+        max_area = 0
+
+        for item in photos:
+            for image_info in item["images"].values():
+                height = image_info["height"]
+                width = image_info["width"]
+                area = height * width
+
+                if area > max_area:
+                    max_area = area
+                    largest_photo_info = {
+                        "url": image_info["url"],
+                        "height": height,
+                        "width": width,
+                        "caption": item.get("caption", ""),
+                        "album": item.get("album", ""),
+                        "published_date": item.get("published_date", ""),
+                        "id": item.get("id", ""),
+                        "source": item.get("source", {}),
+                        "user": item.get("user", {}),
+                    }
+
+        return largest_photo_info
+
+    url = (
+        f"https://api.content.tripadvisor.com/api/v1/location/{location_id}/"
+        f"photos?language={language}&key={api_key}"
+    )
+    headers = {
+        "accept": "application/json",
+    }
+
+    logger.info(f"Requesting photos for location ID {location_id}")
+
+    try:
+        response = requests.get(url, headers=headers)
+        logger.info(
+            f"Received response with status code {response.status_code}",
+        )
+
+        if response.status_code == 200:
+            data = response.json()
+            largest_photo = find_largest_photo(data["data"])
+            logger.info("Successfully retrieved the largest photo")
+            return ServiceResponse(
+                status=ServiceExecStatus.SUCCESS,
+                content={"largest_photo": largest_photo},
+            )
+        error_detail = (
+            response.json()
+            .get("error", {})
+            .get("message", f"HTTP Error: {response.status_code}")
+        )
+        logger.error(f"Error in response: {error_detail}")
+        return ServiceResponse(
+            status=ServiceExecStatus.ERROR,
+            content={"error": error_detail},
+        )
+    except Exception as e:
+        logger.exception("Exception occurred while requesting location photos")
+        return ServiceResponse(
+            status=ServiceExecStatus.ERROR,
+            content={"error": str(e)},
+        )
+
+
+def search_tripadvisor(
+    api_key: str,
+    query: str,
+    language: str = "en",
+) -> ServiceResponse:
+    """
+    Search for locations using the TripAdvisor API.
+
+    Args:
+        api_key (`str`):
+            Your TripAdvisor API key.
+        query (`str`):
+            The search query.
+        language (`str`, optional):
+            The language for the response. Defaults to 'en'.
+
+    Returns:
+        `ServiceResponse`: A dictionary with two variables: `status` and
+        `content`. The `status` variable is from the ServiceExecStatus enum,
+        and `content` is the JSON response from TripAdvisor API or error
```

Comment:
What information is in the return of TripAdvisor API? With this function, users do not know how to deal the content of ServiceResponse.

Reply:
I have added a description of the return data as well as an example in the docstring.